src/cmd/src/cmdUtils.cpp

```diff
@@ -0,0 +1,131 @@
+#include <unistd.h>
+#include <sys/types.h>
+#include <dirent.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "cmdUtils.h"
+
+char * stringcat(char *out, const char *in)
+{
+  char *anchor = out;
+  size_t olen;
+
+  if(out == NULL || in == NULL)
+    return NULL;
+
+  olen = strlen(out);
+  out += olen;
+  while((*out++ = *in++))
+    ;
+  return anchor;
+}
```

Comment:
These functions are all awfully C-like. There are methods to work with strings and paths in C++ that let you do the same thing without having to write any custom algorithms and without introducing buffer overflow errors like here.